lightning/src/chain/channelmonitor.rs

```diff
@@ -948,33 +947,46 @@ impl<ChanSigner: ChannelKeys + Writeable> ChannelMonitor<ChanSigner> {
 
 		assert!(commitment_transaction_number_obscure_factor <= (1 << 48));
 		let our_channel_close_key_hash = WPubkeyHash::hash(&shutdown_pubkey.serialize());
 		let shutdown_script = Builder::new().push_opcode(opcodes::all::OP_PUSHBYTES_0).push_slice(&our_channel_close_key_hash[..]).into_script();
 		let payment_key_hash = WPubkeyHash::hash(&keys.pubkeys().payment_point.serialize());
 		let counterparty_payment_script = Builder::new().push_opcode(opcodes::all::OP_PUSHBYTES_0).push_slice(&payment_key_hash[..]).into_script();
 
-		let counterparty_tx_cache = CounterpartyCommitmentTransaction { counterparty_delayed_payment_base_key: *counterparty_delayed_payment_base_key, counterparty_htlc_base_key: *counterparty_htlc_base_key, on_counterparty_tx_csv, per_htlc: HashMap::new() };
-
-		let mut onchain_tx_handler = OnchainTxHandler::new(destination_script.clone(), keys.clone(), on_holder_tx_csv);
-
-		let holder_tx_sequence = initial_holder_commitment_tx.unsigned_tx.input[0].sequence as u64;
-		let holder_tx_locktime = initial_holder_commitment_tx.unsigned_tx.lock_time as u64;
-		let holder_commitment_tx = HolderSignedTx {
-			txid: initial_holder_commitment_tx.txid(),
-			revocation_key: initial_holder_commitment_tx.keys.revocation_key,
-			a_htlc_key: initial_holder_commitment_tx.keys.broadcaster_htlc_key,
-			b_htlc_key: initial_holder_commitment_tx.keys.countersignatory_htlc_key,
-			delayed_payment_key: initial_holder_commitment_tx.keys.broadcaster_delayed_payment_key,
-			per_commitment_point: initial_holder_commitment_tx.keys.per_commitment_point,
-			feerate_per_kw: initial_holder_commitment_tx.feerate_per_kw,
-			htlc_outputs: Vec::new(), // There are never any HTLCs in the initial commitment transactions
+		let counterparty_channel_parameters = channel_parameters.counterparty_parameters.as_ref().unwrap();
+		let counterparty_delayed_payment_base_key = counterparty_channel_parameters.pubkeys.delayed_payment_basepoint;
+		let counterparty_htlc_base_key = counterparty_channel_parameters.pubkeys.htlc_basepoint;
+		let counterparty_tx_cache = CounterpartyCommitmentTransaction { counterparty_delayed_payment_base_key, counterparty_htlc_base_key, on_counterparty_tx_csv, per_htlc: HashMap::new() };
+
+		let mut onchain_tx_handler = OnchainTxHandler::new(destination_script.clone(), keys.clone(), channel_parameters.clone());
+
+		let secp_ctx = Secp256k1::new();
+
+		let directed_channel_parameters = channel_parameters.to_directed(true);
+		let txid = initial_holder_commitment_tx.calculate_txid(&directed_channel_parameters, &secp_ctx);
+
+		// block for Rust 1.34 compat
+		let (holder_commitment_tx, current_holder_commitment_number) = {
+			let commitment_tx = &initial_holder_commitment_tx.inner;
+			let tx_keys = &commitment_tx.keys;
+			let current_holder_commitment_number = commitment_tx.commitment_number;
+			let holder_commitment_tx = HolderSignedTx {
```

Comment:
Obviously doesn't make sense in this PR, but may be worth adding a TODO to suggest swapping `HolderSignerTx` with a `HolderCommitmentTransaction` + a Vec of `HTLCSource` objects.

Reply:
added on `HolderSignedTx`